cfme/configure/tasks.py

```diff
@@ -49,19 +103,62 @@ def is_task_finished(destination, task_name, expected_status, clear_tasks_after_
     expected_status = re.compile(expected_status, re.IGNORECASE)
 
     try:
-        row = tab_view.table.row(task_name=task_name, state=expected_status)
+        rows = list(tab_view.table.rows(task_name=task_name, state=expected_status))
     except IndexError:
         logger.warn('IndexError exception suppressed when searching for task row, no match found.')
         return False
 
+    # check state = finished for all tasks
+    return expected_num_of_tasks == len(rows), len(rows)
+
+
+def are_all_tasks_analysis_pass(name, expected_num_of_tasks, task_type='container',
+                                silent_failure=False, clear_tasks_after_success=False):
+    """ Check if all tasks analysis match state with no errors"""
+
+    tabs_data = TABS_DATA_PER_PROVIDER[task_type]
+    return all_tasks_analysis_pass(destination=tabs_data['tab'],
```

Comment:
Overall this tasks module needs some enhancements/refactor to actually use BaseCollection/BaseEntity, and any of these methods needs to be implemented (likely as class methods) on the collection class.

That aside, I would much rather avoid adding more 'convenience' wrappers around the various methods for determining task status for a particular type or state.  If we are to have them, I think the naming needs to be more concise. Personally I think this pattern of `all_tasks_<something>_<state>` and `are_all_tasks_<something>_<state>` to be confusing and does not bring anything of value to tests/callers.

This should be a more concise (hopefully just one) method which takes `task_type`, `status`, and `result` kwargs and maps those conditions to the tasks table.

Reply:
Merge the methods to one method , by remove the are_all_tasks_analysis_pass method.
The new Method name is check_tasks_have_no_errors.